konig-workbook-update-maven-plugin/src/main/java/io/konig/maven/google/update/GoogleSheetUpdater.java

```diff
@@ -0,0 +1,1119 @@
+	private static final String DECIMAL_PRECISION = "Decimal Precision";
+	private static final String DECIMAL_SCALE = "Decimal Scale";
+	private static final int HEADER_ROW = 1;
+	private Map<URI,String> prefixList = new HashMap<>();
+	private Map<String,String> targetClassList = new HashMap<>();
+	private ValueFactory vf = new ValueFactoryImpl();
+	
+	private boolean isOverrideSheetValues = false;
+	private boolean isOverrideRDFValues = false;
+	private boolean isRaiseError = false;
+	
+	private  Sheets sheet = null;
+	private  String sheetId = null;
+	
+	public GoogleSheetUpdater(String option) {
+		sheet = getSheetService();
+		if(option.equals("OPTION 1")) {
+			isOverrideSheetValues = true;
+		} else if (option.equals("OPTION 2")) {
+			isOverrideRDFValues = true;
+		} else {
+			isRaiseError = true;
+		} 
+	}
+	
+	public void execute(FileSet[] owlFileSet ,FileSet[] shaclFileSet, String documentId) throws RDFParseException, RDFHandlerException, IOException, IriTemplateParseException {
+		this.sheetId = documentId;
+		for(FileSet fileSet : owlFileSet) {
+			if(fileSet.getDirectory() != null){
+				File viewDir = new File(fileSet.getDirectory());
+				File[] files = viewDir.listFiles();
+				for(File file : files) {
+					try (InputStream inputStream = new FileInputStream(file)) {
+						Graph graph = new MemoryGraph();
+						RdfUtil.loadTurtle(graph, inputStream, "");
+						updateOntology(graph);
+						//updateClasses(graph);
+						//updateProperties(graph);
+					}
+				}
+			}
+		}
+		for(FileSet fileSet : shaclFileSet) {
+			if(fileSet.getDirectory() != null){
+				File viewDir = new File(fileSet.getDirectory());
+				File[] files = viewDir.listFiles();
+				for(File file : files) {
+					try (InputStream inputStream = new FileInputStream(file)) {
+						Graph graph = new MemoryGraph();
+						RdfUtil.loadTurtle(graph, inputStream, "");
+						updateShapes(graph);
+						updatePropertyConstraints(graph);
+					}
+				}
+			}
+		}
+	}
+	
+	private Sheets getSheetService() {
+		String credentialFile = System.getProperty("io.konig.maven.google.update");
+		if(credentialFile == null || credentialFile.equals("")) {
+			throw new KonigException("Credential not found. Please define the 'io.konig.maven.google.update' system property ");
+		}
+		InputStream in = null;
+		try {
+			in = new FileInputStream(new File(credentialFile));
+			GoogleCredential credential = GoogleCredential.fromStream(in).createScoped(DriveScopes.all());
+			sheet = new Sheets.Builder(Utils.getDefaultTransport(), Utils.getDefaultJsonFactory(), credential)
+					.setApplicationName("Google Sheet API")
+					.build();
+		}catch(Exception ex){
+			throw new KonigException(ex);
+		}
+		return sheet;
+	}
+	
+	private void updateOntology(Graph graph) throws IOException {
+		List<Vertex> ontologyList = graph.v(OWL.ONTOLOGY).in(RDF.TYPE).toVertexList();
+		List<List<Object>> ontologies = null;
+		List<Object> headers = new ArrayList<>();
```

Comment:
As a general rule, try to use strongly typed collections.
For instance, if all the elements in a list are Strings, then you should use `List<String>` instead of `List<Object>`.

I think that's the case here, right?